core/src/main/scala/caliban/schema/ArgBuilder.scala

```diff
@@ -205,6 +209,13 @@ trait ArgBuilderInstances extends ArgBuilderDerivation {
 
   implicit lazy val upload: ArgBuilder[Upload] = {
     case StringValue(v) => Right(Upload(v))
-    case other          => Left(ExecutionError(s"Can't build an Upload from $other"))
+    case other          => Left(InvalidInputArgument("Upload", other.toString))
   }
 }
+
+case object InvalidInputArgument extends NoStackTrace {
+  override def getMessage: String = "invalid input argument"
+
+  private[caliban] def apply(expected: String, actual: String): ExecutionError =
```

Comment:
Is it too weird to have the `apply` method defined within the error itself? Mima complains if I create a private method within the `ArgBuilderInstances` for some reason

Reply:
I think it's okay. Missing the word "build" in the message btw